app/src/main/java/com/physical_web/cms/physicalwebcms/DriveSetupActivity.java

```diff
@@ -0,0 +1,196 @@
+        setupButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                startSetup();
+            }
+        });
+
+        // Services API is installed or not installed, only need to check on activity startup
+        setupGoogleAPISnackbar();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        // connectivity can change often, check whenever focus is lost
+        setupConnectivitySnackbar();
+    }
+
+    @Override
+    public void onConnectionSuspended(int cause) {
+    }
+
+    @Override
+    public void onConnected(Bundle connectionHint) {
+        Log.d(MainActivity.TAG, "Drive connection successful");
+
+        unlockInterface();
+
+        Button button = (Button) this.findViewById(R.id.start_setup_button);
+        button.setText("Next");
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                setResult(Activity.RESULT_OK);
+                finish();
+            }
+        });
+        button.setEnabled(true);
+    }
+
+    @Override
+    public void onConnectionFailed(ConnectionResult connectionResult) {
+        if (connectionResult.hasResolution()) {
+            try {
+                connectionResult.startResolutionForResult(this, RESOLVE_CONNECTION_REQUEST_CODE);
+            } catch (IntentSender.SendIntentException e) {
+                // TODO check which cases this exception is encountered
+                throw new RuntimeException("There is no resolution to the Drive API error");
+            }
+        } else {
+            GooglePlayServicesUtil.getErrorDialog(connectionResult.getErrorCode(), this, 0).show();
+        }
+    }
+
+    @Override
+    protected void onActivityResult(final int requestCode, final int resultCode,
+                                    final Intent data) {
+        switch (requestCode) {
+            case RESOLVE_CONNECTION_REQUEST_CODE:
+                if (resultCode == RESULT_OK) {
+                    apiClient.connect();
+                } else {
+                    unlockInterfaceWithWarning();
+                }
+                break;
+        }
+    }
+
+    private void startSetup() {
+        lockInterface();
+        apiClient.connect();
+    }
+
+    // disable button and show spinner to indicate background work
+    private void lockInterface() {
+        busyAuthorizing = true;
+        findViewById(R.id.textViewWarning).setVisibility(View.INVISIBLE);
+        findViewById(R.id.indeterminateBar).setVisibility(View.VISIBLE);
+        findViewById(R.id.start_setup_button).setEnabled(false);
+    }
+
+    // enable interface interaction after a successful authorization
+    private void unlockInterface() {
+        findViewById(R.id.indeterminateBar).setVisibility(View.INVISIBLE);
+        this.findViewById(R.id.textViewWarning).setVisibility(View.INVISIBLE);
```

Comment:
Why you are using `this.findById` and in some cases `findById` (without `this`) ?